### DISCUSSION TITLE
Is that possible to know the line number of element?

### DISCUSSION THREAD

**Discussion by iaomw** [drmohundro/SWXMLHash#139] [Ideas]
Hi, I am using your project as XML parser.  And I also want do some mark up on my UI.
It will be easier if I know the line number. But I didn't find any possible method.

Is that possible to know the line line number of specific XMLIndexer I am dealing with?

**Comment by drmohundro** [comment]
Are you asking if it is possible to know the line number if you were speaking of text?

For example:

```xml
<xml>
  <contents>
    <content>First</content>
    <content>Second</content>
  </contents>
</xml>
```

And then the element with the text of `Second` would have a line number of `4`?

That information isn't provided by the library today; however, it looks like `NSXMLParser` has a `lineNumber` property so it is possible that this functionality could be added. It'd be difficult to know which elements would need to expose this, though.

For example, attributes can be on their own line. Even textual content can be on its own line, but today it is only exposed as a string property.

**Comment by iaomw** [comment]
Thanks, understand your concern. I didn't play with `NSXMLParser`, so not sure how it behaves. The `lineNumber` and `columnNumber` are just `NSInteger`. If that will be added on your `XMLElement`, it's more logical to expose it as a range. 

For example: the line range of `contents` is `(2, 5)`
```
<xml>
  <contents>
    <content>First</content>
    <content>Second</content>
  </contents>
</xml>
```
 Actually, I am trying to mark up some inline attributes and leaf attributes on my UI with the original XML, so 'columnNumber' is also a good thing.

For example: mark the `minimumLineSpacing="10"` and `<size key="itemSize" width="50" height="50"/>`
 

```
<collectionViewFlowLayout key="collectionViewLayout" minimumLineSpacing="10" minimumInteritemSpacing="10" id="Rhn-xg-EmY">
                                    <size key="itemSize" width="50" height="50"/>
                                    <size key="headerReferenceSize" width="0.0" height="0.0"/>
                                    <size key="footerReferenceSize" width="0.0" height="0.0"/>
                                    <inset key="sectionInset" minX="0.0" minY="0.0" maxX="0.0" maxY="0.0"/>
                                </collectionViewFlowLayout>
```

**Comment by drmohundro** [comment]
Yeah, great point having it be modeled as a range - that's spot on. I'm certainly open the possibility of adding it at some point. I'll tag this as a feature request.
